packages/coordinator/src/services/verification/certificate-revocation/utils.ts

```diff
@@ -0,0 +1,76 @@
+import {fromBER} from "asn1js"
+import axios from "axios"
+import {X509} from "jsrsasign"
+import {CertificateRevocationList} from "pkijs"
+import {RevokedCertificate} from "pkijs"
+import {bufferToHexCodes} from "pvutils"
+import {hl7V3} from "@models"
+import {convertHL7V3DateTimeToIsoDateTimeString} from "../../translation/common/dateTime"
+import {extractSignatureDateTimeStamp, getCertificateTextFromPrescription} from "../common"
+
+const CRL_REASON_CODE_EXTENSION = "2.5.29.21"
+
+const getRevokedCertSerialNumber = (cert: RevokedCertificate): string => {
+  const certHexValue = cert.userCertificate.valueBlock.valueHexView
+  return bufferToHexCodes(certHexValue).toLocaleLowerCase()
+}
+
+const getPrescriptionSignatureDate = (parentPrescription: hl7V3.ParentPrescription): Date => {
+  const prescriptionSignedDateTimestamp = extractSignatureDateTimeStamp(parentPrescription)
+  return new Date(convertHL7V3DateTimeToIsoDateTimeString(prescriptionSignedDateTimestamp))
+}
+
+const getCertificateFromPrescription = (parentPrescription: hl7V3.ParentPrescription): X509 => {
+  try {
+    const x509CertificateText = getCertificateTextFromPrescription(parentPrescription)
+    const x509CertificatePem = `-----BEGIN CERTIFICATE-----\n${x509CertificateText}\n-----END CERTIFICATE-----`
+    const x509Certificate = new X509(x509CertificatePem)
+    return x509Certificate
+  } catch (e) {
+    return null
+  }
+}
+
+const wasPrescriptionSignedAfterRevocation = (prescriptionSignedDate: Date, cert: RevokedCertificate): boolean => {
+  const certificateRevocationDate = new Date(cert.revocationDate.value)
+  return prescriptionSignedDate >= certificateRevocationDate
+}
+
+const getRevocationList = async (crlFileUrl: string): Promise<CertificateRevocationList> => {
+  const resp = await axios(crlFileUrl, {method: "GET", responseType: "arraybuffer"})
+  if (resp.status === 200) {
+    const asn1crl = fromBER(resp.data)
+    return new CertificateRevocationList({schema: asn1crl.result})
```

Comment:
Check if pkijs has something to validate the CRL issuer

Reply:
Consider adding this a part of AEA-2998 work

Reply:
PKIjs seems to support it https://pkijs.org/docs/classes/CertificateRevocationList.html#verify

Reply:
If necessary, CRL validation will be done as part of https://nhsd-jira.digital.nhs.uk/browse/AEA-2998